### ISSUE TITLE
Which price do you use in the  Simple Trading Simulation

### ISSUE THREAD

**Issue by halcyone215** [zcakhaa/DeepLOB-Deep-Convolutional-Neural-Networks-for-Limit-Order-Books#3]
'Suppose our model produces a prediction of +1  at time t, we then buy shares at time t+5(taking slippage into account), and hold until -1 appears to sell all u shares (we do nothing if 0 appears).'  
Would you like to tell which price you use at time t+5?  the mid price at t+5 or others? and why do action buy at t+5, not at t?

**Comment by zcakhaa** [comment]
It is the mid-price at t+5. Essentially, this step is used to take latency into account by delaying your enter time point.
You can see better results using the price at t, but in practice, it is unlikely to fill the order at this price.

________________________________

**Comment by halcyone215** [comment]
Thanks a lot for your help. Other questions confuse me are as following. 'hold until -1 appears to sell all u shares'. When and at which price you sell all u shares? at the time predict -1 appears and use the mid price at the time? And what is your train and test step time, a ticket of quotation arrived or the prediction horizon k. Is k=20 means 20 tickets of quotation?

**Comment by zcakhaa** [comment]
Selling is essentially the same as buying. Suppose you get a selling signal at t, we wait for 5 more steps to account for latency and use mid-price at t+5. Time steps are tick time where order books are updated. I am not sure what you mean by the time step in terms of k, but the prediction horizon is defined in the paper.


________________________________

**Comment by halcyone215** [comment]
Thanks for your reply. For example k=20, the model give a buying signal at t, we get u shares, at t+1, another buying signal, ...at t+6,still buying signal,then we will have 6 u shares. at t+7, the model give a selling signal, we sell all 6 u shares .  At time t the model give a buying signal, that is to say, it predict the price will rise at time t+k,i.e t+20,relavive to the price at t. But we do not sell the first u shares at t+20, we sell at t+7.
So I think whether the model predict time step is k, the next step is t+k,not t+1. and then when sell shares, they all past the prediction horizon.

**Comment by zcakhaa** [comment]
I see what you mean. It is a bit different.
Suppose you get a buying signal (+1) at t, you wait until t+5 and use that mid-price as the buying price, then you hold until the -1 appears and then sell your shares (not at t+6). If you get +1 signals either at t+6 or t+7 or ..., there is no point selling your shares when the market moves in favor of you.

________________________________

**Comment by halcyone215** [comment]
yes. As the label is the mean price for the k prediction horizon, not at the t+k time. How do you set the threshold to make the label (-1,0,1)?

**Comment by zcakhaa** [comment]
I am confused. It is a classification problem where we predict +1, 0, or -1. We are not predicting prices. You can also look at https://arxiv.org/abs/2010.13924 for the setup.

________________________________

**Comment by halcyone215** [comment]
Yse,your paper is a classification predict. What I say the mean price for the k prediction horizon, is that using M+(t)the mean price to get the percentage change (lt), not to say your paper is to predict price. Thanks for recommend paper.
